common/tests/cmd_test.go

```diff
@@ -0,0 +1,56 @@
+package tests
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"go.k6.io/k6/cmd"
+	k6tests "go.k6.io/k6/cmd/tests"
+)
+
+func TestCmd(t *testing.T) {
+	t.Skip("data race")
```

Comment:
This test for now is skipped because it causes data race

```
==================
WARNING: DATA RACE
Write at 0x00c0003076d0 by goroutine 24:
  runtime.closechan()
      /home/olegbespalov/.go/src/runtime/chan.go:397 +0x0
  go.k6.io/k6/cmd.(*cmdRun).run.func11()
      /home/olegbespalov/go/pkg/mod/go.k6.io/k6@v0.54.0/cmd/run.go:274 +0xc5
  runtime.deferreturn()
      /home/olegbespalov/.go/src/runtime/panic.go:605 +0x5d
  go.k6.io/k6/cmd.(*cmdRun).run-fm()
      <autogenerated>:1 +0x64
  github.com/spf13/cobra.(*Command).execute()
      /home/olegbespalov/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:856 +0xbbd
  github.com/spf13/cobra.(*Command).ExecuteC()
      /home/olegbespalov/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x617
  github.com/spf13/cobra.(*Command).Execute()
      /home/olegbespalov/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902 +0x24c
  go.k6.io/k6/cmd.(*rootCommand).execute()
      /home/olegbespalov/go/pkg/mod/go.k6.io/k6@v0.54.0/cmd/root.go:108 +0x228
  go.k6.io/k6/cmd.ExecuteWithGlobalState()
      /home/olegbespalov/go/pkg/mod/go.k6.io/k6@v0.54.0/cmd/root.go:137 +0x2b
  github.com/grafana/xk6-browser/common/tests.TestCmd()
      /home/olegbespalov/projects/grafana/k6/xk6-browser/common/tests/cmd_test.go:53 +0x29e
  testing.tRunner()
      /home/olegbespalov/.go/src/testing/testing.go:1690 +0x226
  testing.(*T).Run.gowrap1()
      /home/olegbespalov/.go/src/testing/testing.go:1743 +0x44

Previous read at 0x00c0003076d0 by goroutine 108:
  runtime.chansend()
      /home/olegbespalov/.go/src/runtime/chan.go:171 +0x0
  go.k6.io/k6/metrics.PushIfNotDone()
      /home/olegbespalov/go/pkg/mod/go.k6.io/k6@v0.54.0/metrics/sample.go:135 +0x77d
  github.com/grafana/xk6-browser/common.(*FrameSession).parseAndEmitWebVitalMetric()
      /home/olegbespalov/projects/grafana/k6/xk6-browser/common/frame_session.go:365 +0x543
  github.com/grafana/xk6-browser/common.(*FrameSession).onEventBindingCalled()
      /home/olegbespalov/projects/grafana/k6/xk6-browser/common/frame_session.go:321 +0x26b
  github.com/grafana/xk6-browser/common.(*FrameSession).initEvents.func1()
      /home/olegbespalov/projects/grafana/k6/xk6-browser/common/frame_session.go:309 +0x535

Goroutine 24 (running) created at:
  testing.(*T).Run()
      /home/olegbespalov/.go/src/testing/testing.go:1743 +0x825
  testing.runTests.func1()
      /home/olegbespalov/.go/src/testing/testing.go:2168 +0x85
  testing.tRunner()
      /home/olegbespalov/.go/src/testing/testing.go:1690 +0x226
  testing.runTests()
      /home/olegbespalov/.go/src/testing/testing.go:2166 +0x8be
  testing.(*M).Run()
      /home/olegbespalov/.go/src/testing/testing.go:2034 +0xf17
  main.main()
      _testmain.go:45 +0x164

Goroutine 108 (finished) created at:
  github.com/grafana/xk6-browser/common.(*FrameSession).initEvents()
      /home/olegbespalov/projects/grafana/k6/xk6-browser/common/frame_session.go:239 +0x385
  github.com/grafana/xk6-browser/common.NewFrameSession()
      /home/olegbespalov/projects/grafana/k6/xk6-browser/common/frame_session.go:148 +0xa07
  github.com/grafana/xk6-browser/common.NewPage()
      /home/olegbespalov/projects/grafana/k6/xk6-browser/common/page.go:307 +0xc97
  github.com/grafana/xk6-browser/common.(*Browser).onAttachedToTarget()
      /home/olegbespalov/projects/grafana/k6/xk6-browser/common/browser.go:317 +0x5c8
  github.com/grafana/xk6-browser/common.(*Browser).initEvents.func1()
      /home/olegbespalov/projects/grafana/k6/xk6-browser/common/browser.go:234 +0x464
==================
--- FAIL: TestCmd (1.02s)
    testing.go:1399: race detected during execution of test
FAIL
FAIL    github.com/grafana/xk6-browser/common/tests     1.137s
```

Reply:
I'm wondering if that's the issue itself, keeping in mind that data race is coming from the places which haven't part of the diff

Reply:
This is strange. So it looks like we're trying to write a metric, but at the same time the `state.Sample` is being closed. I would have expected the context to have been done before the `sample` was closed so that it is not written to the channel in `PushIfNotDone`.

Reply:
If i comment out [this](https://github.com/grafana/k6/blob/feb4ac9d42527638c1df17a3e6b36216ea8ffb31/metrics/sample.go#L135), the test you wrote works. Is the race condition in k6? Happy to pair on this debugging session as it will help me understand the lifecycle of k6.

Reply:
Yes, it works, but it reports about the data race if you run the tests with the race detection

```
go test -timeout 30s -race -run ^TestCmd$ github.com/grafana/xk6-browser/common/tests
```

Yep, let's pair on that! I've double-checked and it's actually report of data race (same) for other browser scripts input :cry: